tests/native/scaled.cpp

```diff
@@ -1,14 +1,37 @@
+#include "gtest/gtest.h"
```

Comment:
I also tried adding a test case for `std:complex`
```
  TEST(scaled, mdspan_complex)
  {
    using vector_element_type = std::complex<double>;
    using scaling_factor_type = std::complex<double>;

    using vector_t =
    mdspan<vector_element_type, extents<dynamic_extent>>;

    constexpr std::size_t vectorSize (5);
    constexpr std::size_t storageSize = vectorSize;
    std::vector<vector_element_type> storage (storageSize);

    vector_t x (storage.data (), vectorSize);

    for (std::size_t k = 0; k < vectorSize; ++k) {
      const vector_element_type x_k = vector_element_type(k) + 1.0;
      x(k) = x_k;
    }   

    const scaling_factor_type scalingFactor (-3.0);
    auto x_scaled = scaled (scalingFactor, x); 

    using std::abs;

    for (std::size_t k = 0; k < vectorSize; ++k) {
      auto abs_x_scaled_k = abs(x_scaled(k));
    }   
  }
```
and it ran into the compilation error that I had before (I needed to add a `static_cast` get this to work)
```
/home/yu/repos/nv/stdblas_test/stdBLAS/tests/native/scaled.cpp:42:44: error: no matching function for call to ‘abs(std::experimental::mdspan<const std::complex<double>, std::experimental::extents<18446744073709551615>, std::experimental::layout_right, std::experimental::__p1673_version_0::linalg::accessor_scaled<std::complex<double>, std::experimental::default_accessor<std::complex<double> > > >::reference)’
   42 |       auto abs_x_scaled_k = abs(x_scaled(k));
      |                                            ^
```

Reply:
@youyu3 Let me take a look at this one.  Thanks for reviewing and for the test case!

Reply:
@youyu3 In general, I won't be able to make `scaled_scalar` a replacement for its underlying `value_type` in every possible expression.  I can add a hidden friend `abs` that makes this particular expression work for unqualified `abs`.  That seems reasonable, given that the result of `scaled` could be the input of `vector_norm_2`.  What do you think?

Reply:
Yeah that sounds reasonable to me if there is no better solution.  `scaled` could be the input of a few other blas1 functions as well.  Also we might need this treatment for other functions that can be called with both non-complex and complex data types, like `exp` and `pow`, although they are not used in  stdblas functions themselves?

Reply:
@youyu3 wrote:
> Also we might need this treatment for other functions that can be called with both non-complex and complex data types, like exp and pow, although they are not used in stdblas functions themselves?

The result of `scaled`, `conjugated`, `transposed`, or `conjugate_transposed` only needs to be usable inside stdblas functions.  It's an interesting side effect that it happens to behave in a certain way outside stdblas functions, but that was never the goal of these functions.  We do need `abs` to work generally because of `vector_abs_norm`.

Reply:
@youyu3 I've added `abs`, `real`, and `imag` to `proxy_reference`.  This should make `vector_abs_sum` work.